prow/cmd/sinker/main.go

```diff
@@ -218,13 +221,14 @@ func (c *controller) clean() {
 		}
 		maxPodAge := c.config().Sinker.MaxPodAge
 		for _, pod := range pods.Items {
-			if _, ok := isFinished[pod.ObjectMeta.Name]; !ok {
-				// prowjob is not marked as completed yet
+			if !isFinished.Has(pod.ObjectMeta.Name) && isExist.Has(pod.ObjectMeta.Name) {
```

Comment:
We can be stricter, right? If the Pod exists, is `created-by-prow`, and does not have a non-terminated ProwJob associated with it at the moment, we can remove it.

Reply:
sure

Reply:
made the pod clean decision more explicit here